Data Structures/Linked-Lists/Circular Linked List/deletion.cpp

```diff
@@ -0,0 +1,52 @@
+/* Deletes a node of given address in a circular linked list where last node's next points to first node.
+   There are 3 cases: 1. node to be deleted is the last node
+                      2. node to be deleted is the only node in list.
+                      3. node to be deleted is present at the beginning of list
+   
+   To run this code, make sure you make appropriate calls to this function in main().
+*/
+
+
+struct node {
+ 
+  int data;
+  node *next;
+};
+
+/* Deletes a node of given address
+    node** - address of the HEAD (beginning of linked list)
+    given_node* - node to be deleted(its address is passed as argument)
+*/
+void del(node **head, node *given_node){
+  
+  //checks if the node is the very first node of list
+  if((*head) == given_node){
+    //to check if it is the only node in the list
+    if((*head)->next == NULL)
+      *head = NULL;
+    else
+      *head = given_node->next;
+  }
+  else{
+    node *temp = *head;
+    while(temp->next == given_node)
+      temp = temp->next;
+    //check if given_node is at the end
+    if(given_node->next == (*head))
+      temp->next = *head;
+    //if it's present in between two nodes
+    else
+      temp->next = given_node->next;
+  }
+ 
+  delete given_node;
+  return;
+ 
+}
+  
+/* Time Complexity : O(n) if node is present at the end or is present between two nodes.
+                     O(1) if node is present at the beginning.
```

Comment:
Time complexity is a general thing so it should only be O(n). Mention single line explanation for O(n) time complexity.